File: JavaLab/src/main/java/ch2/section2/sub2/Quiz1.java
package ch2.section2.sub2;

public class Quiz1 {
    public static void main(String[] args) {
        int i = 4;
        if (i>=3 && i<=5) {
            System.out.println("봄입니다.");
        } else if (i>=6 &&i<=8) {
            System.out.println("여름입니다.");
        } else if (i>=9 &&i<=11) {
            System.out.println("가을입니다.");
        } else if (i==12 || i==1 || i==2){
            System.out.println("겨울입니다.");
        } else {
            System.out.println("알 수 없는 계절입니다.");
        }

    }
}

File: KotlinLab/KotlinLab/src/main/kotlin/ch2/section7_nullsafety/q3/Quiz1.kt
package org.example.ch2.section7.q3

class Address(var state: String?, var city: String?)

class User(val name: String, var address: Address?)

class Quiz1 {

    fun printUserInfo(user: User?) {
        if(user == null) {
            println("유저 정보가 없습니다.")
            return
        }
        val addressInfo = user.address ?: throw NullPointerException("주소 없음")
        println("${user.name}: ${addressInfo.state} ${addressInfo.city}")
    }
}

fun main() {
    Quiz1().printUserInfo(null)
    val user = User("홍길동", null)
    val address = Address(null,null)

    address.city = "성남시"
    address.state = "경기도"
    user.address = address
    println(Quiz1().printUserInfo(user))
}